shell/shell.js

```diff
@@ -16,22 +16,131 @@
 
 const { classes: Cc, interfaces: Ci, results: Cr, utils: Cu } = Components;
 const { Runtime } = Cu.import('resource://qbrt/modules/Runtime.jsm', {});
+const { Services } = Cu.import('resource://gre/modules/Services.jsm', {});
 
-window.addEventListener('load', event => {
-  const browser = document.getElementById('content');
-  const url = window.arguments[0];
+function Shortcuts() {
+  this.isMac = Services.appinfo.OS === 'Darwin';
+  this.keys = {
+    i: 73,
+    r: 82,
+    f5: 116,
+  };
+}
+Shortcuts.prototype.reloadPage = function(event) {
+  dump(`reloadPage ${this.isMac} ${event.metaKey} ${event.ctrlKey} ` +
+       `${event.keyCode === this.keys.r} ${event.keyCode}\n`);
+  if (this.isMac) {
+    // `Cmd + R` or `F5`.
+    return (event.metaKey && event.keyCode === this.keys.r) || event.keyCode === this.keys.f5;
+  }
+  // `Ctrl + Shift + R`.
+  return event.ctrlKey && event.keyCode === this.keys.r;
+};
+Shortcuts.prototype.hardReloadPage = function(event) {
+  if (!event.shiftKey) {
+    return false;
+  }
+  dump(`hardReloadPage ${this.isMac} ${event.shiftKey} ${event.metaKey} ` +
+       `${event.ctrlKey} ${event.keyCode === this.keys.r} ${event.keyCode}\n`);
+  if (this.isMac) {
+    // `Cmd + R` or `F5`.
+    return (event.metaKey && event.keyCode === this.keys.r) || event.keyCode === this.keys.f5;
+  }
+  // `Ctrl + Shift + R`.
+  return event.ctrlKey && event.keyCode === this.keys.r;
+};
+Shortcuts.prototype.toggleDevTools = function(event) {
+  dump(`openDevTools ${this.isMac} ${event.metaKey} ${event.altKey} ` +
+       `${event.keyCode === this.keys.i}\n`);
+  if (this.isMac) {
+    // `Cmd + Alt + I`.
+    return event.metaKey && event.altKey && event.keyCode === this.keys.i;
+  }
+  // `Ctrl + Shift + I`.
+  return event.ctrlKey && event.shiftKey && event.keyCode === this.keys.i;
+};
+Shortcuts.prototype.openDevTools = Shortcuts.prototype.toggleDevTools;
+Shortcuts.prototype.closeDevTools = Shortcuts.prototype.toggleDevTools;
 
-  browser.loadURI(url, null, null);
-  // dump instead of console.log to write to stdout for tests.
-  dump(`opened ${url} in new window\n`);
-  Runtime.openDevTools(browser);
+window.addEventListener('load', () => {
+  UI.init();
+}, { once: true });
 
-  browser.addEventListener('keydown', event => {
-    // Reload the web page when the F5 key is pressed.
-    if (event.keyCode && event.keyCode === 116 && !event.shiftKey &&
-        !event.altKey && !event.ctrlKey && !event.metaKey) {
-      browser.reload();
-    }
-  }, false, true);
+window.addEventListener('unload', () => {
```

Comment:
Whoops, I missed this question. It may be a benign error, but it's worth having an issue on file, so please do file it!